### DISCUSSION TITLE
Testing ARM64 SBC's - 0.83.0 alpha builds w/Nuked-SC55 CLAP Plugin

### DISCUSSION THREAD

**Discussion by hubby4** [dosbox-staging/dosbox-staging#4135] [General]
Testing ARM64 SBC's with 0.83.0 development builds and the Nuked-SC55 CLAP plugin with games that support Sound Canvas and General MIDI.

This thread is a continuation of posts I made here. [LINK](https://github.com/dosbox-staging/dosbox-staging/pull/4090#issuecomment-2558460047)

**Raspberry Pi 5**

OS: Raspberry Pi OS 64-bit (Based on Debian 12 - Bookworm)
Desktop Resolution: 3840x2160

The Raspberry Pi 5 2gb model can be bought for $50 USD.

I built both dosbox-staging and the Nuked-SC55 CLAP plugin using Meson.

In the .bashrc system startup file I use the following settings.

```
export SDL_AUDIODRIVER=pipewire
```

Raspberry Pi OS uses the wayland display compositor and pipewire audio by default. I've opted to use the default SDL2 x11 driver because the SDL2 wayland driver drops frames more than just running the SDL2 x11 driver through xwayland.

In the dosbox-staging.conf file I changed the following settings to avoid occasional audio distortion.

```
vsync=off
presentation_mode=cfr
```

Peformance is often too slow with the default crt-auto opengl shader. So I've replaced it with the fakelottes.tweaked shader from [LINK](https://github.com/dosbox-staging/dosbox-shaders/tree/master/crt). 

With Duke Nukem 3D and Blood, I was able to run both games with good performance at 800x600 and 640x480 SVGA resolution with cpu_cycles_protected=max.

**Raspberry Pi 5: Overclock Settings**

OS: Raspberry Pi OS 64-bit (Based on Debian 12 - Bookworm)
Desktop Resolution: 3840x2160

Thanks to the silicon lottery, recent audio distortion improvements by weirddan455 and johnnovak, and further tweaking, I can now run games up to 25000 cycles (486 DX2-66mhz) with the default crt-auto shader and the Nuked-SC55 CLAP plugin.

My overclock settings in the /boot/firmware/config.txt

```
#25% CPU Overclock
arm_freq=3000
#25% GPU Overclock
gpu_freq=1000
force_turbo=1
over_voltage_delta=50000
```

In the .bashrc system startup file I use the following settings.

`export SDL_AUDIODRIVER=pipewire`

I no longer run the SDL2 wayland driver since it degrades performance. The default SDL2 x11 driver running through xwayland results in less frame drops.

In the dosbox-staging.conf file I changed the following settings from their defaults.

```
host_rate=30.000
vsync=off
presentation_mode=cfr
cpu_cycles_protected=25000
blocksize=1024
prebuffer=25
```

Audio distortion and pops/clicks are gone. Frame and audio dropouts rarely occur. I haven't seen any major drawbacks forcing a 30FPS host_rate so far.

Duke Nukem 3D and Dark Forces are playable at 320x200 VGA. Blood is too slow, even at 320x200 VGA resolution. Not surprising since the minimum requirement is a Pentium 75mhz.

Overall, I satisfied with the performance with the overclock. I've played games for hours without stability issues so far.

**Orange Pi 5**

OS: Ubuntu 22.04 64-bit (Custom Build) - [LINK](https://github.com/Joshua-Riek/ubuntu-rockchip)
Desktop Resolution: 3840x2160

The Orange Pi 5 4gb model can be found on Amazon US for $80 USD. About the price of an 8gb Raspberry Pi 5.

I built both dosbox-staging and the Nuked-SC55 CLAP plugin using Meson.

In the .bashrc system startup file I use the following settings.

```
export SDL_VIDEODRIVER=wayland
export SDL_AUDIODRIVER=pulseaudio
```

The Ubuntu 22.04 OS uses the wayland display compositor and pulseaudio audio by default.

The hacked GPU driver this OS ships with requires using the PAN_MESA_DEBUG=gofaster environment variable to achieve the best performance in opengl games and emulators. I find that it's better to do this when starting up dosbox-staging, since this environment variable sometimes causes system instability when doing non-gaming tasks like web browsing.

In the folder with the dosbox staging executable I create a dosbox.sh bash script with the following text.

```
#!/bin/bash
export PAN_MESA_DEBUG=gofaster
./dosbox
```

From the folder open the terminal and type:

`chmod +x dosbox.sh`

This will make the script executable.

When running dosbox-staging using the bash script, the Orange Pi 5 can use the crt-auto shader without suffering the performance penalties that the Raspberry Pi 5 has. No changes to the dosbox-staging.conf file are necessary.

Similar to the Raspberry Pi 5, I ran both Duke Nukem 3D and Blood with good performance at 800x600 and 640x480 SVGA resolution with cpu_cycles_protected=max.

**Comment by johnnovak** [comment]
Thanks for all this useful info @hubby4. Great to hear Staging works well on the Pi 5 with the Nuked SC55 CLAP plugin 🎉

**Comment by hubby4** [comment]
I've updated the OP with settings for overclocking the Raspberry Pi 5 for those who don't mind voiding their warranty.
